compiler/src/main/kotlin/ast/ast.kt

```diff
@@ -76,27 +75,22 @@ data class InverseExpr(val ctx: ParseTree, val value: Expr) : Expr()
 
 data class ArrayLookupExpr(val ctx: ParseTree, val ident: String = MAGIC_UNDEFINED_STRING, val index: Int = -1) : Expr()
 
-data class ArrayBodyExpr(val ctx: ParseTree, val type: String?, val values: List<Expr>) : Expr()
+data class ArrayBodyExpr(val ctx: ParseTree, val type: String? = MAGIC_UNDEFINED_STRING, val values: List<Expr>) : Expr()
```

Comment:
I think we should make types to an enum.

Reply:
I'm not sure enums are the right solution, how would you represent an array of integers?